src/lib/optimizer/strategy/chunk_pruning_rule.cpp

```diff
@@ -173,12 +181,12 @@ std::vector<PredicatePruningChain> ChunkPruningRule::_find_predicate_pruning_cha
    *      started, we pass an empty vector.
    *   3. The third argument refers to the StoredTableNode for which predicate pruning chains should be returned.
    *      Therefore, we have to pass stored_table_node again.
-   *   4. The fourth argument is used to track the visited nodes. Predicate pruning chains can branch out, but are not
-   *      expected to merge again. This is checked by tracking the visited nodes.
+   *   4. The fourth argument is used for debugging purposes: Predicate pruning chains can branch out,
+   *      but are not expected to merge again. Therefore, some visited nodes are tracked.
```

Comment:
"some" is not really helpful.

Reply:
Changed to 

```
*   4. The fourth argument is used for debugging purposes: Predicate pruning chains can branch out,
   *      but are not expected to merge again. Therefore, all nodes that are visited as part of a predicate chain are
   *      tracked.
```

Reply:
For me, the semantics of "tracking" a node are still unclear. Are we only tracking PradicateNodes? If not, what does "visited as part of a predicate chain" mean?

Reply:
I agree, it is confusing. I was not happy with this either. 

I adjusted the Assert to check for revisited PredicateNodes only. It should be much clearer now. Please have another look.

Reply:
If you could add a sentence on *why* that would be illegal (i.e., it is an assumption that we make to prevent our recursion from visiting nodes twice), the only thing between us and the merge are the benchmark results.

Reply:
```
   *   4. The fourth argument is used for debugging purposes: The following function's recursion should not lead to a
   *      repeated visitation of nodes. Due to the assumption that predicate pruning chains do not merge after 
   *      having branched out, visited predicate nodes are tracked and checked for revisitation in an Assert.
```

ok?

Reply:
Sound good.